dash-renderer/src/observers/requestedCallbacks.ts

```diff
@@ -0,0 +1,263 @@
+import {
+    all,
+    concat,
+    difference,
+    filter,
+    flatten,
+    groupBy,
+    intersection,
+    isEmpty,
+    isNil,
+    map,
+    values
+} from 'ramda';
+
+import { IStoreState } from "../store";
+
+import {
+    aggregateCallbacks,
+    removeRequestedCallbacks,
+    removePrioritizedCallbacks,
+    removeExecutingCallbacks,
+    removeWatchedCallbacks,
+    addRequestedCallbacks,
+    addPrioritizedCallbacks,
+    addExecutingCallbacks,
+    addWatchedCallbacks
+} from '../actions/callbacks';
+
+import { isMultiValued } from '../actions/dependencies';
+
+import {
+    combineIdAndProp,
+    getReadyCallbacks,
+    getUniqueIdentifier,
+    pruneCallbacks
+} from '../actions/dependencies_ts';
+
+import {
+    ICallback,
+    IExecutingCallback,
+    IStoredCallback
+} from '../types/callbacks';
+
+import { getPendingCallbacks } from "../utils/callbacks";
+import { IStoreObserverDefinition } from '../StoreObserver';
+
+const observer: IStoreObserverDefinition<IStoreState> = {
+    observer: ({
+        dispatch,
+        getState
+    }) => {
+        const { callbacks, callbacks: { prioritized, executing, watched, executed, stored }, paths } = getState();
+        let { callbacks: { requested } } = getState();
+
+        const pendingCallbacks = getPendingCallbacks(callbacks);
```

Comment:
This observer promotes `requested` callbacks to `prioritized` for later execution. It handles most pruning cases upfront although some additional pruning does happen in the prioritized and executing observers.

Reasons for removing/modifying callbacks are numerous and documented below.